src/main/java/software/amazon/awssdk/crt/mqtt/MqttConnection.java

```diff
@@ -180,9 +204,9 @@ public void onFailure(String reason) {
             future.complete(null);
             return future;
         }
-        AsyncCallback disconnectAck = wrapVoidFuture(future);
+        disconnectAck = wrapVoidFuture(future);
         connectionState = ConnectionState.DISCONNECTING;
```

Comment:
Doesn't need to be fixed in this CR, but I notice that this class doesn't protect any of its internal state with locks. So multiple threads calling `disconnect()` are going to scramble things like disconnectAck

Reply:
Luckily, connectionState is volatile, and it basically doesn't matter atm, but you're right, I've been actively ignoring this for a while, needs a pass soon.